examples_v30x/crc_test/ch32v30x_crc.h

```diff
@@ -0,0 +1,34 @@
+#include "ch32fun.h"
+
+void CRC_init(void) {
+	// Enable CRC clock
+	RCC->AHBPCENR |= RCC_AHBPeriph_CRC;
+}
+
+// Calculate CRC for a single 32-bit value
+u32 CRC_calculate32(u32 data) {
+	// Reset CRC unit
+	CRC->CTLR = CRC_CTLR_RESET;
+	
+	// Write data to trigger CRC calculation
+	CRC->DATAR = data;
+	
+	// Wait for CRC calculation
```

Comment:
This is impractical.  Please add the while(...) to wait for the completion.  And verify that it works.